### ISSUE TITLE
Source links for instance methods don't point at the right source line

### ISSUE THREAD

**Issue by sjakobi** [haskell/haddock#665]
In the shown link from [the haddocks here](http://hackage.haskell.org/package/these-0.7.4/docs/Control-Monad-Chronicle-Class.html#t:MonadChronicle) the correct URL fragment would be `#line-128`.

![bildschirmfoto vom 2017-08-06 22-43-02](https://user-images.githubusercontent.com/3664523/29006903-2eca3d94-7af9-11e7-89fb-52084c82258c.png)

**Comment by alexbiehl** [comment]
Yes, this is known. I am not sure why it is the way it is currently.

**Comment by harpocrates** [comment]
Could you confirm this is now fixed? I just tried regenerating the docs for `these` using the `haddock` that came with GHC 8.4.2 and I could not reproduce. Here is what I tried:

    $ cabal new-haddock --with-ghc /usr/local/bin/ghc-8.4.2 \
                        --with-haddock /usr/local/bin/haddock-ghc-8.4.2 \
                        --haddock-for-hackage
    $ tar xpvf dist-newstyle/these-0.7.4-docs.tar.gz
    $ open these-0.7.4-docs/Control-Monad-Chronicle-Class.html  # ...then clicked source link

**Comment by sjakobi** [comment]
> Could you confirm this is now fixed?

No. I'm still directed to the definition of the `dictate` method instead of `MaybeT`'s implementation of `dictate`.
